modules/test/playwright/tests/frontend-data-set-views-web/pages/DataSetsPage.ts

```diff
@@ -102,28 +105,16 @@ export class DataSetsPage {
 	async goto() {
 		await this.applicationsMenuPage.goToDataSetManager();
 
-		await this.page
-			.getByRole('heading', {
-				name: 'Data Sets',
-			})
-			.waitFor();
+		await this.pageContainer.waitFor();
```

Comment:
@juanjofgliferay I believe you noticed the same issue in https://github.com/liferay-frontend/liferay-portal/pull/4002, that check for `heading` is not sufficient. You added
```js
await page.locator('.dnd-table > .dnd-thead > .dnd-tr').waitFor();
```
This does basically the same thing, but also provides a handy container reference, for easier locators. I added several of these container locators in PR, even adding CSS classes in code to enable them.

Reply:
@markocikos Yep, I come to the same conclusion. This is something we need to do at this point, otherwise we will end up using `waitFor()` across the tests